### PR TITLE
VSRE-414: Get TeamCity CI working

### PR DESCRIPTION
**Pull request by mkostick**
This gets TeamCity CI working again on the version of this gem's code from #4.  We've decided that all the changes since then should be undone.  #6 has already been reverted, but #5 couldn't be done in isolation.  It was opened, after all, because the code from #4 no longer passed.

I structured this into two commits.  The first is a revert of #5.  The second is the set of test/development fixes to get TeamCity CI passing.  There are no changes to the actual gem code, so this does not need to be published as a new version.  Details:

- Update to a recent version of Ruby

- Use some Ruby `prepend` magic to normalize the `Thrift:ThinHTTPServer::RackApplication`
  response.  In most cases, a Rack app can return either an array of `[status, header, body]` or a
  `Rack::Response` that encapsulates the three.  The former is technically the only form permitted
  by the Rack specification, while the latter is what's returned by `Thrift:ThinHTTPServer`.  This
  wasn't a problem when `faraday_http_client_transport_spec.rb` was originally written, but since
  then `Rack::Lint` has been updated to allow only the array response.  I don't know why
  `Thrift:ThinHTTPServer` hasn't been updated to fix this, but this workaround is the simplest
  solution.

  See https://github.com/apache/thrift/blob/1d6a3262cf32d5063cfcb9ee09355aa1315e7f80/lib/rb/lib/thrift/server/thin_http_server.rb#L73
  for the server code and https://github.com/rack/rack/commit/6ab28c22b065b19b45b068f030b84a8bdb89b145
  for the linter change.

- Regenerate the thrift files for the test with the latest Thrift version.  The old version was
  causing issues because it was expecting to receive a payload containing just the serialzed object,
  while what was actually being sent had a header that needed to be read first.  This can be seen in
  the diff for `YetiThriftTest::SimpleService::Client#recv_mutate`.

  Command used: `thrift -o spec/support --gen rb -I thrift spec/thrift/spec.thrift`

- Include `active_support` explicitly in a test that now needs it.

I intentionally did not try to add cloudbuild support back in.  That can be done as a follow-on

### CODE REVIEW

**Comment by mkostick** [comment]
@vendasta/sre @vendasta/yw-staff-senior-devs

**Comment by mkostick** [comment]
The last commit allows newer versions of `rspec` (3.x instead of 3.0.x) to work with the latest version of `rake`.

**Comment by mkostick** [comment]
Grrr... still not there.  CI installed `rack` 3.x, but I have 2.x installed locally.  Newer versions of `thin` restrict their `rack` version to 2.x.  Old ones were naive and just required >1, so CI included some ancient one along with rack 3.x

I'm going to try adding a development dependency for `rack`.

**Comment by mkostick** [comment]
OK, this is ready for review - CI is green 💚

File: spec/lib/gem_ext/thrift/transport/faraday_http_client_transport_spec.rb [hunk 1]
```diff
@@ -1,6 +1,33 @@
...
+    def failed_request
+      unpack_response(super)
+    end
+  end
+end
+
+
+class Thrift::ThinHTTPServer::RackApplication
+  class << self
+    prepend ResponseNormalizer::ClassMethods
```
Comment (avadan1) [inline spec/lib/gem_ext/thrift/transport/faraday_http_client_transport_spec.rb:27]: Neat! Wasn't aware of this Ruby feature.

**Review by avadan1 — APPROVED** [review]

**Review by rafalyesware — APPROVED** [review]
LGTM, thanks!

**Pull request merged**